sp/src/game/client/ez2/c_npc_wilson.cpp

```diff
@@ -230,6 +253,99 @@ C_RopeKeyframe *C_NPC_Wilson::CreateRope( int iStartAttach, int iEndAttach )
 	return pRope;
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: 
+// Input  : *classes - 
+//			phoneme - 
+//			scale - 
+//			newexpression - 
+//-----------------------------------------------------------------------------
+void C_NPC_Wilson::AddViseme( Emphasized_Phoneme *classes, float emphasis_intensity, int phoneme, float scale, bool newexpression )
+{
+	int type;
+
+	// Setup weights for any emphasis blends
+	bool skip = SetupEmphasisBlend( classes, phoneme );
+	// Uh-oh, missing or unknown phoneme???
+	if ( skip )
+	{
+		return;
+	}
+		
+	// Compute blend weights
+	ComputeBlendedSetting( classes, emphasis_intensity );
+
+	for ( type = 0; type < NUM_PHONEME_CLASSES; type++ )
+	{
+		Emphasized_Phoneme *info = &classes[ type ];
+		if ( !info->valid || info->amount == 0.0f )
+			continue;
+
+		const flexsettinghdr_t *actual_flexsetting_header = info->base;
+		const flexsetting_t *pSetting = actual_flexsetting_header->pIndexedSetting( phoneme );
+		if (!pSetting)
+		{
+			continue;
+		}
+
+		flexweight_t *pWeights = NULL;
+
+		int truecount = pSetting->psetting( (byte *)actual_flexsetting_header, 0, &pWeights );
+		if ( pWeights )
+		{
+			float flPhonemes = npc_wilson_talk_start.GetFloat();
+			for ( int i = 0; i < truecount; i++)
+			{
+				// Translate to global controller number
+				//int j = FlexControllerLocalToGlobal( actual_flexsetting_header, pWeights->key );
+				flPhonemes += info->amount * scale * pWeights->weight * npc_wilson_talk_scale.GetFloat();
+				// Go to next setting
+				pWeights++;
+			}
+
+			// Lerp it
+			m_flTalkGlow = FLerp( npc_wilson_talk_min.GetFloat(), npc_wilson_talk_max.GetFloat(), flPhonemes / m_flTalkGlow );
+		}
+	}
+}
+
+class CWilsonEyeMaterialProxy : public CResultProxy
+{
+public:
+	CWilsonEyeMaterialProxy()
+	{
+	}
+	virtual ~CWilsonEyeMaterialProxy()
+	{
+	}
+	virtual bool Init( IMaterial *pMaterial, KeyValues *pKeyValues )
+	{
+		if (!CResultProxy::Init( pMaterial, pKeyValues ))
+			return false;
+
+		return true;
+	}
+	virtual void OnBind( void *pC_BaseEntity )
+	{
+		if (!pC_BaseEntity)
+			return;
+
+		C_BaseEntity *pEntity = BindArgToEntity( pC_BaseEntity );
+
+		// HACKHACK: The Wilson model may be used for dynamic props instead, so make sure it's actually him
+		if (FClassnameIs( pEntity, "npc_wilson" ))
+		{
+			C_NPC_Wilson *pWilson = static_cast<C_NPC_Wilson*>(pEntity);
+			SetFloatResult( pWilson->m_flTalkGlow );
+		}
```

Comment:
Might be a good idea to set a constant value when used on non-npc props, so it's not dependent on whatever the glow was set to in the last frame?

Reply:
With how it works now, wouldn't the result variable just remain at its default value if it's used on non-NPC props? Or is it better practice for the result to always be set to something regardless of conditions?

Reply:
It should always set the value. Material parameters only get assigned from the VMT when it's initially loaded, after that the game expects the proxies to override previous values from other renders. If you had both an NPC and non-NPC prop in the same level currently, the non-NPC would end up either using the NPC's values or what it had the previous frame, depending on the render order. Which would be very weird and camera-dependent.